src/openzaak/components/catalogi/models/mixins.py

```diff
@@ -119,3 +119,32 @@ class Meta:
     def publish(self):
         self.concept = False
         self.save()
+
+
+class BlockRelationToPublishedMixin:
+    """
+    Mixin for models related to a specified resource, to ensure that new relations
+    non-concept instances of this resource are blocked
+    """
+
+    resource_name = ""
+
+    def clean(self):
+        super().clean()
+
+        # Related resource not present in data, do not attempt to check for concept
+        if not getattr(self, f"{self.resource_name}_id"):
+            return
+
+        if not getattr(self, self.resource_name).concept:
+            raise ValidationError(
+                {
+                    self.resource_name: _(
+                        "Creating a relation to non-concept {resource_name} is forbidden"
+                    ).format(resource_name=self.resource_name)
+                }
+            )
```

Comment:
Can't we implement that as a validator on the FK field instead?

```py
class StatusType(...):
    zaaktype = models.ForeignKey("ZaakType", ..., validators=[validate_concept_status])
```

Reply:
You're right, I wasn't sure about this mixin approach